keras/utils/dataset_utils.py

```diff
@@ -14,14 +14,328 @@
 """Keras image dataset loading utilities."""
 
+
+
+
+
 import tensorflow.compat.v2 as tf
 # pylint: disable=g-classes-have-attributes
 
 import multiprocessing
 import os
+import time
+import warnings
+from random import Random
 
 import numpy as np
+from tensorflow.python.util.tf_export import keras_export
+
+@keras_export('keras.utils.split_dataset')
+def split_dataset(dataset, 
+                  left_size=None, 
+                  right_size=None, 
+                  shuffle=False, 
+                  seed=None):
+  """Split a dataset into a left half and a right half (e.g. training / validation).
+  
+  Args:
+      dataset: A `tf.data.Dataset` object or 
+               a list/tuple of arrays with the same length.   
+      left_size: If float, it should be in range `[0, 1]` range 
+                 and signifies the fraction of the data to pack in 
+                 the left dataset. If integer, it signifies the number 
+                 of samples to pack in the left dataset. 
+                 If `None`, it defaults to the complement to `right_size`.       
+      right_size: If float, it should be in range `[0, 1]` range 
+                  and signifies the fraction of the data to pack 
+                  in the right dataset. If integer, it signifies 
+                  the number of samples to pack in the right dataset.
+                  If `None`, it defaults to the complement to `left_size`.     
+      shuffle: Boolean, whether to shuffle the data before splitting it.
+      seed: A random seed for shuffling.
+
+  Returns:
+      A tuple of two `tf.data.Dataset` objects: the left and right splits.
+  """
+  
+  if not isinstance(dataset,(tf.data.Dataset,list,tuple,np.ndarray)):
+    raise TypeError('`dataset` must be either a tf.data.Dataset object'
+                   f' or a list/tuple of arrays. Received : {type(dataset)}')
+    
+  if right_size is None and left_size is None:
+    raise ValueError('you must specify either `left_size` or `right_size`'
+                    ' Received: `left_size`= None, and `right_size`=None')
+      
+  dataset_as_list = _convert_dataset_to_list(dataset)
+  
+  if seed is None:
+    seed = np.random.randint(1e6)  
+    
+  if shuffle: 
+    Random(seed).shuffle(dataset_as_list)
+    
+  total_length = len(dataset_as_list)
+   
+  left_size,right_size = _rescale_dataset_split_sizes(left_size,
+                                                      right_size,
+                                                      total_length)
+  
+  left_split = list(dataset_as_list[:left_size])
+  right_split = list(dataset_as_list[-right_size:])
+
+  if _get_type_spec(dataset) is tuple:
+    left_split = tuple(np.array(sample) for sample in zip(*left_split))
+    right_split = tuple(np.array(sample) for sample in zip(*right_split))
+  
+  try:
+    left_split = tf.data.Dataset.from_tensor_slices(left_split)
+  except Exception as e:
```

Comment:
Please except a specific exception -- not a generic Exception. What's the exception that will be raised here?